libs/utils/android/workloads/viewer.py

```diff
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from time import sleep
+
+from android import Screen, System
+from android.workload import Workload
+
+
+class Viewer(Workload):
+    """
+    Android generic Viewer workload
+    """
+    
+    # Let the system pick the best package
+    package = ''
+
+    def __init__(self, test_env):
+        super(Viewer, self).__init__(test_env)
+        self._log = logging.getLogger('Viewer')
+        self._log.debug('Workload created')
+
+        # Set of output data reported by the browser
+        self.db_file = None
+
+    def run(self, out_dir, uri, workload, collect=''):
+        """
+        Run viewer workload
+
+        :param out_dir: Path to experiment directory where to store results.
+        :type out_dir: str
+
+        :param uri: The URI of the item to display
+        :type location_search: str
+
+        :param workload: Function pointer to the workload to run once item
+        is opened and being viewed.
+        :type workload: func
+
+        :param collect: Specifies what to collect. Possible values:
+            - 'energy'
+            - 'systrace'
+            - 'ftrace'
+            - any combination of the above
+        :type collect: list(str)
+        """
+
+        # Keep track of mandatory parameters
+        self.out_dir = out_dir
+        self.collect = collect
+
+        # Set min brightness
+        Screen.set_brightness(self._target, auto=False, percent=0)
+        # Unlock device screen (assume no password required)
+        Screen.unlock(self._target)
+
+        # Force screen in PORTRAIT  mode
+        Screen.set_orientation(self._target, portrait=True)
+
+        System.gfxinfo_reset(self._target, self.package)
+        sleep(1)
+
+        # Open the requested uri
+        System.view_uri(self._target, uri)
+
+        # Let the activity start
+        sleep(3)
```

Comment:
Same here... I would hide the sleep in the above call in such a way to make it kind-of a sync call.
In the future we can make it more robust by checking (maybe via logcat) if the activity has been really started... but that's definitively for another possible future improvement.